File: server/job/src/main/java/io/iamcyw/ams/job/strategy/persistence/StrategyPushPO.java
package io.iamcyw.ams.job.strategy.persistence;

import io.iamcyw.ams.domain.job.strategy.model.StrategyPushDO;
import io.iamcyw.ams.domain.job.strategy.model.StrategyPushMetaDO;
import io.quarkus.hibernate.orm.panache.PanacheEntity;

import javax.persistence.*;
import java.util.Map;
import java.util.Set;
import java.util.stream.Collectors;

@Entity(name = "ALARM_STRATEGY_PUSH")
public class StrategyPushPO extends PanacheEntity {

    public String subject;

    public String type;

    public String template;

    @OneToMany(fetch = FetchType.LAZY)
    public Set<StrategyPushMetaPO> meta;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "STRATEGY_ID")
    public AlarmStrategyPO strategy;

    public StrategyPushDO as() {
        return new StrategyPushDO(id, subject, template, asMeta());
    }

    public StrategyPushMetaDO asMeta() {
        Map<String, String> metas = meta.stream().collect(
                Collectors.toMap(metaPO -> metaPO.metaKey, metaPO -> metaPO.metaValue));
        return new StrategyPushMetaDO(type, metas);
    }

}

File: domain/src/jvmMain/kotlin/io/iamcyw/ams/domain/job/strategy/StrategyPushPO.kt
package io.iamcyw.ams.domain.job.strategy

import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntityBase
import javax.persistence.Entity

@Entity(name = "ALARM_STRATEGY_PUSH")
actual class StrategyPushPO : PanacheEntityBase {

    companion object : PanacheCompanion<StrategyPushPO>

    /**
     * 消息主题，可以包含表达式
     */
    actual var subject: String = ""

    actual var type: String = ""

    /**
     * 正文模板，可以包含表达式
     */
    actual var template: String = ""

    actual var metaPO: MutableSet<StrategyPushMetaPO>? = null

}